### PR TITLE
PP-8188 Serialise `gateway_account_credentials` for frontend routes

### PR DESCRIPTION
**Pull request by sfount**
Serialise gateway account credentials entity properties.

`/v1/frontend/accounts/{id}` and
`/v1/frontend/accounts/external-id/{id}` should include the full
crednetials list. This shouldn't be included on other routes.

```json
"gateway_account_credentials": [
    {
      "version": 1,
      "payment_provider": "worldpay",
      "credentials": {
        "password": "a-password",
        "merchant_id": "a-merchant-id",
        "username": "a-username"
      },
      "state": "CREATED",
      "last_updated_by_user_external_id": null,
      "created_date": "2021-06-04T17:34:44.080Z",
      "active_start_date": null,
      "active_end_date": null,
      "gateway_account_credential_id": 1
    }
  ],
```

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceIT.java [hunk 1]
```diff
@@ -72,6 +72,10 @@ public void shouldGetGatewayAccountForExistingAccount() {
                 .body("credentials.username", is(gatewayAccountPayload.getUserName()))
                 .body("credentials.password", is(nullValue()))
                 .body("credentials.merchant_id", is(gatewayAccountPayload.getMerchantId()))
+                .body("gateway_account_credentials.size()", is(1))
+                .body("gateway_account_credentials[0].credentials.merchant_id", is(gatewayAccountPayload.getMerchantId()))
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceIT.java:76]: Can we remove password from credentials and assert that `password` is excluded ?

File: src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceIT.java [hunk 2]
```diff
@@ -72,6 +72,10 @@ public void shouldGetGatewayAccountForExistingAccount() {
...
+                .body("gateway_account_credentials[0].payment_provider", is("worldpay"))
+                .body("gateway_account_credentials[0].state", is("CREATED"))
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceIT.java:78]: PR comment indicates that `gateway_account_id` is not in the `gateway_account_credentials` ? and we can ignore `version` for serialisation

Comment (sfount) [reply src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountFrontendResourceIT.java:78]: 🤔  yeah I thought this was strange, other responses driven by entities also include `version` (gateway account, charge, etc.) so I just left it in there

**Comment by sfount** [comment]
Thanks @kbottla, password redacted from external response using a similar methodology to other routes and added a specific getter for the account ID to match the spec https://github.com/alphagov/pay-connector/pull/2964/commits/54d43a6221cd85575367d28964169f70e0da322e

**Review by kbottla — APPROVED** [review]
LGTM!

**Pull request merged**